src/client/components/Forms/PasswordResetForm.tsx

```diff
@@ -0,0 +1,108 @@
+import React from "react";
+import styles from "./Form.module.css";
+import { useCustomTranslation } from "@coin-view/client";
+import cx from "classnames";
+import { signOut } from "next-auth/react";
+import { useRouter } from "next/router";
+
+export const PasswordResetForm = (token: any) => {
+  const { t, language } = useCustomTranslation();
+  const newPasswordRef = React.useRef<HTMLInputElement>(null);
+  const repeatNewPasswordRef = React.useRef<HTMLInputElement>(null);
+
+  const { push, locale } = useRouter();
+
+  const [error, setError] = React.useState();
+
+  const checkPasswordValidity = React.useCallback(() => {
+    const password = newPasswordRef.current?.value;
+    const confirmpassword = repeatNewPasswordRef.current?.value;
+    if (password !== confirmpassword && repeatNewPasswordRef.current) {
+      repeatNewPasswordRef.current?.setCustomValidity("Passwords don't match!");
+    } else {
+      repeatNewPasswordRef.current?.setCustomValidity("");
+    }
+  }, [newPasswordRef, repeatNewPasswordRef]);
+
+  const submitForm = React.useCallback(
+    async (evt: React.FormEvent<HTMLFormElement>) => {
+      evt.preventDefault();
+      const form = evt.target as HTMLFormElement;
+      if (!form.checkValidity()) {
+        return;
+      }
+
+      const newPassword = newPasswordRef.current?.value.trim();
+      const repeatNewPassword = repeatNewPasswordRef.current?.value.trim();
+
+      const response = await fetch("/api/auth/resetpassword", {
```

Comment:
I think we should have a function that will reduce this boilerplate code or use React query. Please add a task to replace usage of every client side fetch with one of those solutions